### ISSUE TITLE
3.0.0 nomap has internet permission again

### ISSUE THREAD

**Issue by serrq** [IacobIonut01/Gallery#471]
The previously version (2.1.2) was without internet permission.

![1000118349](https://github.com/user-attachments/assets/1e7cdba4-bdf1-4ca9-8787-9480c66502c2)

**Comment by IacobIonut01** [comment]
The no-maps ci removes the internet permission from the manifest before building, I'm currently investigating which dependencies are adding it back for some reason

**Comment by IacobIonut01** [comment]
> Okay thanks.
> 
> Could I have the nomap edition on F-droid too?
> 
> Currently I switched from 2.1.1 (fdroid) to 3.0.0 nomap on Github.

I don't control which edition is on f-droid. The f-droid team builds it automatically when a new release happens.

However, the internet permission should only be used to download a map preview of any image that has gps metadata embedded

**Comment by IacobIonut01** [comment]
> Okay for your assurances. But if you can remove it at all for me it is better.

That's my goal, the no-map to not have any internet permissions. I'm still confused why it has popped back, anyway, good catch.

**Comment by IacobIonut01** [comment]
> How I can write about it to Fdroid team?

You can try contacting them using the closed issue about f-droid in this repo. However I don't think there's a chance for them to alter the automation of builds for a single app that requires an extra script run pre-build

**Comment by IzzySoft** [comment]
[Removing Unwanted Manifest Permissions With tools:node](https://developermemos.com/posts/removing-unwanted-manifest-permissions/) comes to mind, so it would not need an extra script to remove it.

> I don't think there's a chance for them to alter the automation of builds for a single app that requires an extra script run pre-build

Should be easy, and they already have it (since v2.0.0 btw):

```yaml
  - versionName: 3.0.0
    versionCode: 30033
    commit: 1eefc8705963fc43916c44583f179265162af490
    subdir: app
    gradle:
      - yes
    prebuild: sed -i -e '/INTERNET/d' src/main/AndroidManifest.xml
```

> Could Izzy integrating the nomap edition in his repo?

Depends on. There are no per-ABI APKs attached to releases. And seeing the UniversalAPK having 126 MB I very much doubt any single ABI will have an APK much smaller than 40 MB. At IoD, there's a per-app size limit of 30 MB (with very few exceptions).

**Comment by IacobIonut01** [comment]
Should be fixed now in the 3.0.1-30102 nightly release

**Comment by IacobIonut01** [comment]
Verified and it's fixed
